torch/nn/parallel/distributed.py

```diff
@@ -614,6 +627,174 @@ def train(self, mode=True):
             module.train(mode)
         return self
 
+    # When running in join mode, schedules an allreduce to match the one in the
+    # forward pass to determine the no. of currently active processes and whether
+    # all processes have joined.
+    def _schedule_shadow_all_reduce_for_fwd_pass(self):
+        all_active_procs = torch.zeros(
+            1, device=self.device_ids[0] if self.device_type != "cpu" else "cpu"
+        )
+        dist.all_reduce(all_active_procs, group=self.process_group)
+        return all_active_procs.item()
+
+    # When running in join mode, checks and performs sync of module buffers if
+    # the models have buffers that should be synchronized in the forward pass.
+    def _check_and_sync_module_buffers(self):
+        if self.will_sync_module_buffers():
+            my_rank = dist.get_rank(self.process_group)
+            rank_to_use = self._find_common_rank(my_rank, False)
```

Comment:
other places call this `authoritative_rank`. Either sounds good to me, but let's keep them consistent.